### PR TITLE
VACMS-13138: 508 defect--Search fields should have the landmark role="search"

### PR DESCRIPTION
**Pull request by dsasser**
## Summary

Resolves a 508 Defect 2 by:

1. Using the [Search Input DS component](https://design.va.gov/components/search-input), which is already accessible
2. Adds a landmark role to the wrapping div.

### Consequences
1. Because the Search Input component is a DS web component, and the input and button elements are now in the shadow DOM, a couple unit tests were removed which were targeting specific input elements. However, the cypress tests cover the functionality so we are not losing test confidence.
2. The form styling has change slightly, as the form elements are styled using the DS component styling.
3. Form _error_ styling has changed. The search box will no longer have a red border. See below:

Error Display Before:
Screenshot 2023-05-19 at 9 16 02 AM

Error Display After:
Screenshot 2023-05-19 at 11 05 50 AM

## Related issue(s)
Relates to #13138

## Testing done
Manual testing to confirm expected behavior

## Screenshots
|         | Before | After |
| ------- | ------ | ----- |
| Mobile  |    https://github.com/department-of-veterans-affairs/vets-website/assets/221539/d993b3cf-a0e1-435e-b43d-b23b784a3640    |    https://github.com/department-of-veterans-affairs/vets-website/assets/221539/cb364b96-630a-49e3-9795-eda926042a43   |
| Desktop |    https://github.com/department-of-veterans-affairs/vets-website/assets/221539/52455fbe-d683-4a10-aef6-ad15c9c219c4    |    https://github.com/department-of-veterans-affairs/vets-website/assets/221539/088fcc66-cd05-48c9-85e7-15eec8c2b33f   |

## What areas of the site does it impact?
/find-forms

### Quality Assurance & Testing
- [x] I fixed|updated|added unit tests and integration tests for each feature.
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x] Linting warnings have been addressed
- [x] Screenshot of the developed feature is added
- [x] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [x] Browser console contains no warnings or errors.

### CODE REVIEW

File: src/applications/find-forms/containers/SearchForm.jsx [hunk 1]
```diff
@@ -76,82 +78,51 @@ export const SearchForm = ({ fetchForms, fetching }) => {
...
   );
 };
 
 SearchForm.propTypes = {
-  // From mapStateToProps.
-  fetching: PropTypes.bool.isRequired,
   // From mapDispatchToProps.
   fetchForms: PropTypes.func.isRequired,
+  // From mapStateToProps.
+  fetching: PropTypes.bool.isRequired,
```
[the remark this answers is not in the document]
Comment (dsasser) [reply src/applications/find-forms/containers/SearchForm.jsx]: Good catch; fixed!

**Review by randimays — COMMENTED** [review]
There's a spacing issue on narrower windows; the error state of the form gets a bit cut off because it's aligned with the left edge of the window.

Local:
Screenshot 2023-05-30 at 10 58 04 AM

Staging:
Screenshot 2023-05-30 at 10 58 09 AM

File: src/applications/find-forms/containers/SearchForm.jsx [hunk 2]
```diff
@@ -76,80 +78,47 @@ export const SearchForm = ({ fetchForms, fetching }) => {
   };
 
   return (
-    <form
+    <div
+      role="search"
+      aria-label="Searh VA Forms"
```
[the remark this answers is not in the document]
Comment (dsasser) [reply src/applications/find-forms/containers/SearchForm.jsx]: Good catch! Fixed.

**Comment by dsasser** [comment]
> There's a spacing issue on narrower windows; the error state of the form gets a bit cut off because it's aligned with the left edge of the window.

Nice catch! Fixed.

**Comment by laflannery** [comment]
There currently is no way for the hint text to be programmatically associated with the search input so non-visual users are most likely going to miss that helper text.  With that in mind, can you change the `aria-label` to "Search for a VA form by keyword, form name, or form number."

**Review by randimays — APPROVED** [review]

**Comment by dsasser** [comment]
> There currently is no way for the hint text to be programmatically associated with the search input so non-visual users are most likely going to miss that helper text. With that in mind, can you change the `aria-label` to "Search for a VA form by keyword, form name, or form number."

Done, thanks for the review Laura!

**Comment by thejordanwood** [comment]
@dsasser In the After screenshot, the spacing for the error text looks too far from the label. I believe the spacing should be 12px (1.2rem) but it looks 24px (2.4rem) right now.

**Comment by dsasser** [comment]
> @dsasser In the After screenshot, the spacing for the error text looks too far from the label. I believe the spacing should be 12px (1.2rem) but it looks 24px (2.4rem) right now.

Good catch Jordan! I've updated things to remove some hard-coded padding under the label, so things are looking closer to prod/staging today.

**Pull request merged**